### ISSUE TITLE
When I have addfinalizer in fixture, no violation happens

### ISSUE THREAD

**Issue by graingert** [m-burst/flake8-pytest-style#50]
# Bug report
re-disallow addfinalizer in factories as fixtures pattern:
## What's wrong

I have
```python
# "Factories as fixtures" pattern
@pytest.fixture()
def my_factory(request):
    def create_resource(arg):
        resource = acquire_resource(arg)
        request.addfinalizer(resource.release)
        return resource
    return create_resource
```

## How it should work

it should report the addfinalizer call in favour of yield fixtures:

```python
# "Factories as fixtures" pattern
@pytest.fixture()
def my_factory(request):
    with contextlib.ExitStack() as stack:
        def create_resource(arg):
            resource = acquire_resource(arg)
            stack.callback(resource.release)
            return resource

        yield create_resource
```

or even better:

```python
@contextlib.contextmanager
def _cm_resource(arg):
    resource = acquire_resource(arg)
    try:
        yield resource
    finally:
        resource.release()

# "Factories as fixtures" pattern
@pytest.fixture()
def my_factory(request):
    with contextlib.ExitStack() as stack:
        def create_resource(arg):
            return stack.enter_context(_cm_resource(arg))

        yield create_resource
```

## System information

* Operating system: NA
* Python version: NA
* flake8 version: NA
* flake8-pytest-style version: NA

**Comment by m-burst** [comment]
Hi @graingert,

Thanks for your report!

I decided to specifically allow `addfinalizer` in "factories as fixtures" pattern (see #46). I personally believe that it is a valid use case, and the workaround via `ExitStack` appears more complicated and less readable to me.
